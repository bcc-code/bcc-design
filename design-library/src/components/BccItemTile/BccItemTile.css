@layer components {
  .bcc-item-tile {
    height: 125px;
    @apply flex min-w-full flex-col justify-between gap-y-4 px-3 py-2;
    @apply overflow-y-hidden rounded-lg border border-on-primary text-left shadow;
  }
  .bcc-item-tile-overline {
    @apply text-caption truncate text-secondary;
  }
  .bcc-item-tile-title {
    @apply text-heading truncate font-bold;
  }
}
