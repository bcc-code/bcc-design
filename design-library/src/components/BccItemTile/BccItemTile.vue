<script setup lang="ts">
type Props = {
  overline?: string;
  title?: string;
};

withDefaults(defineProps<Props>(), {});
</script>

<template>
  <div class="bcc-item-tile">
    <p class="text-caption opacity-60">{{ overline }}</p>
    <h2 class="text-heading mb-0.5 truncate font-bold">{{ title }}</h2>
    <slot name="infoRight" />
  </div>
</template>
