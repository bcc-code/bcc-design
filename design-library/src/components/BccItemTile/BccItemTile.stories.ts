import { Face2Icon } from "@bcc-code/icons-vue";
import BccItemTile from "./BccItemTile.vue";

import type { Meta, StoryFn } from "@storybook/vue3";

export default {
  title: "Components/BccItemTile",
  component: BccItemTile,
  argTypes: {},
} as Meta<typeof BccItemTile>;

const Template: StoryFn<typeof BccItemTile> = (args) => ({
  components: { BccItemTile },
  setup() {
    return { args };
  },
  template: `
    <BccItemTile v-bind="args" />
  `,
});

export const Example = Template.bind({});
Example.args = {
  title: "Title",
  overline: "Overline",
};

export const WithInfoOnRight: StoryFn<typeof BccItemTile> = (args) => ({
  components: { BccItemTile, Face2Icon },
  setup() {
    return { args };
  },
  template: `
    <div class="flex flex-col space-y-4">
      <BccItemTile v-bind="args">
        <template #infoRight>
          <div>
            <div class="text-caption-sm flex items-center gap-x-1 text-tertiary">
              <Face2Icon class="h-4 w-4" />
              <p class="truncate">12 Apr 2000</p>
            </div>
            <div class="text-caption-sm flex items-center gap-x-1 text-tertiary">
              <Face2Icon class="h-4 w-4" />
              <p class="truncate">13 Apr 2000</p>
            </div>
          </div>
        </template>
      </BccItemTile>
    </div>
  `,
});

WithInfoOnRight.args = {
  overline: "Overline",
  title: "Title",
};
