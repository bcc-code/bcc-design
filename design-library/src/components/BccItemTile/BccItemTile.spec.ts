import { describe, it, expect } from "vitest";

import { mount } from "@vue/test-utils";
import BccItemTile from "./BccItemTile.vue";

describe("BccItemTile", () => {
  it("renders title and overline", async () => {
    const wrapper = mount(BccItemTile, { props: { title: "Title", overline: "Overline" } });
    expect(wrapper.html()).toContain("Title</h2>");
    expect(wrapper.html()).toContain("Overline</p>");
  });

  it("renders info slot", async () => {
    const wrapper = mount(BccItemTile, {
      props: { title: "Title" },
      slots: {
        default: "<p>Slot markup</p>",
      },
    });
    expect(wrapper.html()).toContain("Title</h2>");
    expect(wrapper.html()).toContain("<p>Slot markup</p>");
  });
});
