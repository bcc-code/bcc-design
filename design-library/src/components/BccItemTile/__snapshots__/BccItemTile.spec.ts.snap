// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BccItemTile > renders info slot 1`] = `
"<div class=\\"bcc-item-tile\\">
  <div>
    <p class=\\"text-caption truncate text-secondary\\"></p>
    <h2 class=\\"text-heading truncate font-bold\\">Title</h2>
  </div>
  <p>Slot markup</p>
</div>"
`;

exports[`BccItemTile > renders title and overline 1`] = `
"<div class=\\"bcc-item-tile\\">
  <div>
    <p class=\\"text-caption truncate text-secondary\\">Overline</p>
    <h2 class=\\"text-heading truncate font-bold\\">Title</h2>
  </div>
</div>"
`;
